src/libs/actions/BankAccounts.js

```diff
@@ -216,12 +226,65 @@ function validateBankAccount(bankAccountID, validateCode) {
     });
 }
 
+/**
+ * @param {Object} data
+ * @returns {Object}
+ */
+function mergeWithLocalACHData(data) {
+    const updatedACHData = {
+        ...reimbursementAccountInSetup,
+        ...data,
+
+        // This param tells Web-Secure that this bank account is from NewDot so we can modify links back to the correct
+        // app in any communications. It also will be used to provision a customer for the Expensify card automatically
+        // once their bank account is successfully validated.
+        enableCardAfterVerified: true,
+    };
+
+    if (data && !_.isUndefined(data.isSavings)) {
+        updatedACHData.isSavings = Boolean(data.isSavings);
+    }
+    if (!updatedACHData.setupType) {
+        updatedACHData.setupType = updatedACHData.plaidAccountID
+            ? CONST.BANK_ACCOUNT.SETUP_TYPE.PLAID
+            : CONST.BANK_ACCOUNT.SETUP_TYPE.MANUAL;
+    }
+
+    return updatedACHData;
+}
+
+/**
+ * Updates the bank account in the database with the company step data
+ *
+ * @param {Object} params
+ * @param {String} [params.companyName]
+ * @param {String} [params.addressStreet]
+ * @param {String} [params.addressCity]
+ * @param {String} [params.addressState]
+ * @param {String} [params.addressZipCode]
+ * @param {String} [params.companyPhone]
+ * @param {String} [params.website]
+ * @param {String} [params.companyTaxID]
+ * @param {String} [params.incorporationType]
+ * @param {String} [params.incorporationState]
+ * @param {String} [params.incorporationDate]
+ * @param {Boolean} [params.hasNoConnectionToCannabis]
+ */
+function updateCompanyInformationForBankAccount(params) {
+    API.write(
+        'UpdateCompanyInformationForBankAccount',
+        mergeWithLocalACHData(params),
```

Comment:
hmmm ... if we go this route (using mergeWithLocalACHData), I'd prefer we also pass an array of keys, so only required params are sent in the request, not along with everything from reimbursementAccountInSetup.

Reply:
I like that idea. I refactored this PR to account for this. Thanks for the suggestion!